machina/traj/traj.py

```diff
@@ -278,6 +278,64 @@ def random_batch(self, batch_size, epoch=1, indices=None, return_indices=False):
                     batch_size, indices, return_indices)
                 yield batch
 
+    def random_batch_rnn(self, batch_size, seq_length=None, epoch=1):
+        """
+        Providing sequences of batch which is randomly sampled from trajectory.
+        batch shape is (seq_length, batch_size, *)
+
+        Parameters
+        ----------
+        batch_size : int
+        seq_length : int
+            Length of sequence of batch.
+        epoch : int
+
+        Returns
+        -------
+        batch : dict of torch.Tensor
+        """
+
+        if seq_length is None:
+            seq_length = max([self._epis_index[i+1] - self._epis_index[i]
+                              for i in range(len(self._epis_index)-1)])
+
+        for _ in range(epoch):
+            seqs = []
+            lengths = []
+            indices = np.random.randint(
+                0, len(self._epis_index)-1, (batch_size,))
+
+            for idx in indices:
+                length = min(
+                    self._epis_index[idx+1] - self._epis_index[idx], seq_length)
+                start = np.random.randint(
+                    self._epis_index[idx], self._epis_index[idx+1] - length + 1)
+                data_map = dict()
+                for key in self.data_map:
+                    if idx == len(self._epis_index)-2 and length != seq_length:
```

Comment:
if seq_length is larger than the sum of the lengths of last two episodes, indexing of Line 319 will be failed. I think `if self._epis_index[-1] - self._epis_index[idx] < seq_length:` may be correct solution.